code/game/objects/items/weapons/implants/implant_death_alarm.dm

```diff
@@ -18,55 +18,45 @@
 	return dat
 
 /obj/item/implant/death_alarm/Destroy()
-	STOP_PROCESSING(SSobj, src)
+	UnregisterSignal(imp_in, COMSIG_MOB_DEATH)
 	return ..()
 
-/obj/item/implant/death_alarm/process()
-	if(!implanted)
-		return
-	var/mob/M = imp_in
-
-	if(isnull(M)) // If the mob got gibbed
-		activate()
-	else if(M.stat == DEAD)
-		activate("death")
-
-/obj/item/implant/death_alarm/activate(cause)
+/obj/item/implant/death_alarm/activate(name, cause) // Death signal sends name followed by the gibbed / not gibbed check
 	var/mob/M = imp_in
 	var/area/t = get_area(M)
 
 	var/obj/item/radio/headset/a = new /obj/item/radio/headset(src)
 	a.follow_target = M
 
 	switch(cause)
-		if("death")
-			if(is_type_in_typecache(t, stealth_areas))
-				//give the syndies a bit of stealth
-				a.autosay("[mobname] has died in Space!", "[mobname]'s Death Alarm")
-			else
-				a.autosay("[mobname] has died in [t.name]!", "[mobname]'s Death Alarm")
+		if(TRUE)
```

Comment:
I don't like this, please make a proc for the callback that checks for the `gibbed` parameter of the signal for clarity. You can then get rid of the `name` argument and just use the `mobname` var